Expand MongoDB types documentation (#927)
hasura/ddn-docs: docs/reference/connectors/mongodb/connector-types.mdx

| `regex`               | regular expression                                                                              |
| `javascript`          | JavaScript code                                                                                 |
| `javascriptWithScope` | JavaScript code with bindings for closure variables                                             |
| `minKey`              | defined to be less than any other value for comparisons, regardless of type                     |
| `maxKey`              | defined to be greater than any other value for comparisons, regardless of type                  |
| `undefined`           | deprecated by MongoDB                                                                           |
| `dbPointer`           | deprecated by MongoDB, not supported by connector                                               |
| `symbol`              | deprecated by MongoDB                                                                           |

:::tip Scalar type names in GraphQL

The table above gives names for types as they are used in connector configuration which is taken from names in BSON

> documentation. On the other hand GraphQL uses upper-case type names. So in your GraphQL schema and in metadata

Reviewer: ```suggestion
documentation. On the other hand, GraphQL uses uppercase type names. In your GraphQL schema and in metadata
```

> configuration in `.hml` files you will see these type names with the first letter upper-cased.

Reviewer: ```suggestion
configuration in `.hml` files you will see these type names with the first letter uppercased.
```

:::

## Complex Types

> The complex types that the connector recognizes are,

Reviewer: ```suggestion
The complex types that the connector recognizes are:
```

| Type Constructor | Details                                                                       |
| -----            | -------                                                                       |
| array            | homogenous array                                                              |
| object           | record type with a fixed set of string keys, and a distinct type for each key |
| nullable         | values of a nullable type might be `null`                                     |

For details on how to write complex types see [Type Expressions](./type-expressions.mdx).

## Representations in JSON

> GraphQL operates using JSON so to interact with BSON data in your database the connector must convert inputs (such as

Reviewer: ```suggestion
GraphQL operates using JSON; to interact with BSON data in your database, the connector must convert inputs (such as
```

filter arguments or mutation inputs) from JSON to BSON, and must convert result data from BSON to JSON. These
conversions are determined by the types configured in your connector's schema.

:::warning Extended JSON

The `extendedJSON` type represents all possible BSON values. Fields of this type are converted to JSON differently from
fields with more specific types. See [Extended JSON][].

[Extended JSON]: /reference/connectors/mongodb/extended-json

:::

> In general conversions between JSON and BSON are symmetric: input values will look the same as response data. Here are

Reviewer: ```suggestion
In general, conversions between JSON and BSON are symmetric: input values will look the same as response data. Here are
```
